src/services/utilities.ts

```diff
@@ -436,6 +436,37 @@ namespace ts {
         return isInCommentHelper(sourceFile, position, /*predicate*/ undefined);
     }
 
+    export function isInJsxAttribute(sourceFile: SourceFile, position: number) {
+        let token = getTokenAtPosition(sourceFile, position);
+
+        if (token && token.kind === SyntaxKind.CloseBraceToken && token.parent.kind === SyntaxKind.JsxExpression) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * returns true if the position is in between the open and close elements of an JSX expression.
+     */
+    export function isInsideJsxElement(sourceFile: SourceFile, position: number) {
+        let token = getTokenAtPosition(sourceFile, position);
+
+        // make a distinction if we're dealing with an empty expression or a non-empty expression i.e
+        // <div>|</div vs <div>Hello | World</div> , where | is current cursor
+        if (token && (token.kind === SyntaxKind.JsxText ||
+            (token.kind === SyntaxKind.LessThanToken && token.parent.kind === SyntaxKind.JsxText))) {

```

Comment:
Can you elaborate on why you need to do this in a comment (as discussed offline)?